### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by Asherslab**
So i know there's already a PR for this, but i decided to make my own Dockerfile. The only thing you need todo (which i think i can fix after i make this first PR) is `docker exec -t -i {docker container} php artisan p:user:make`
the env vars i've made (So far) are:
```
DB_HOST
DB_DATABASE
DB_USERNAME
DB_PASSWORD
```

### CODE REVIEW

**Comment by schrej** [comment]
We are trying to keep the project root directory as clean as possible, could you move the files used in the docker container (entrypoint and webserver conf etc.) to `.dev/docker`?

File: .dev/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  cat /app/.env.example > /app/.env
+fi
+
+if [ "$(cat .env | grep APP_KEY)" == "APP_KEY=" ]; then
+  echo "Generating New Key"
+  php artisan key:generate --force
+fi
+
+echo "4"
+php artisan migrate --force
```
Comment (DaneEveritt) [inline .dev/entrypoint.sh]: `php artisan migrate --seed --force`

**Comment by DaneEveritt** [comment]
Please make the requested changes so this can be approved. :)

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
+FROM alpine:3.7
```
Comment (TBK) [inline Dockerfile]: Please use 3.8 instead.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+COPY . ./
+
+COPY .dev/default.conf /etc/nginx/conf.d/default.conf
+
+RUN cp .env.example .env
+RUN composer install --no-dev
+
+EXPOSE 80
+EXPOSE 443
```
Comment (TBK) [inline Dockerfile]: Why not write it as one line?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    echo "https://repos.php.earth/alpine/v3.7" >> /etc/apk/repositories && \
+    apk add --no-cache --update certbot nginx dcron curl tini php7.2 php7.2-bcmath php7.2-common php7.2-dom php7.2-fpm php7.2-gd php7.2-mbstring php7.2-openssl php7.2-zip php7.2-pdo php7.2-phar php7.2-json php7.2-pdo_mysql php7.2-session php7.2-ctype php7.2-tokenizer php7.2-zlib php7.2-simplexml && \
+    curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
+
+COPY . ./
+
+COPY .dev/default.conf /etc/nginx/conf.d/default.conf
+
+RUN cp .env.example .env
+RUN composer install --no-dev
```
Comment (TBK) [inline Dockerfile]: Each time an instructional command is used a new layer is created, try to minimize the use - https://docs.docker.com/develop/develop-images/dockerfile_best-practices/

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,25 @@
+FROM alpine:3.7
+
+WORKDIR /app
+
+RUN apk add --no-cache wget ca-certificates && \
+    wget -O /etc/apk/keys/phpearth.rsa.pub https://repos.php.earth/alpine/phpearth.rsa.pub && \
+    echo "https://repos.php.earth/alpine/v3.7" >> /etc/apk/repositories && \
+    apk add --no-cache --update certbot nginx dcron curl tini php7.2 php7.2-bcmath php7.2-common php7.2-dom php7.2-fpm php7.2-gd php7.2-mbstring php7.2-openssl php7.2-zip php7.2-pdo php7.2-phar php7.2-json php7.2-pdo_mysql php7.2-session php7.2-ctype php7.2-tokenizer php7.2-zlib php7.2-simplexml && \
```
Comment (TBK) [inline Dockerfile]: Alpine Linux 3.8 got PHP 7.2 packages (worked on it myself).

If you want the latest PHP 7.2 version then I suggest that this uses the official PHP image as the base - https://hub.docker.com/_/php/

**Comment by Asherslab** [comment]
i'm not in the position to develop this any further, apologies. 
But feel free to do with what is there as you will

**Comment by lancepioch** [comment]
@Asherslab Thanks for the contributions.

@TBK Thanks for the review.

**Pull request closed without merging**